### PR TITLE
Add contribution guideline

### LINKED ISSUE #138: Add docs on how to do translations
There should be `contribution.md` file that explains how to do translations and how to submit PR's for this.

Follow up on https://github.com/matterpoll/matterpoll/pull/133

Blocked by #139

### PR DESCRIPTION
**Pull request by hanzei**
This is a first draft for contribution guidelines. Any feedback is welcome!

@jasonblais If you have the time to share feedback, it would be appreciated.

Fixes #138
See https://github.com/matterpoll/matterpoll/pull/133 for details

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+If you think you found a bug in Hugo, [please use the GitHub issue tracker](https://github.com/matterpoll/matterpoll/issues/new?template=bug_report.md) to open an issue. When opening an issue, please provide the required information in the issue template.
+
+
+## Helping with translations
+
+Matterpoll supports localization in various languages. Because we as the maintainer only speak a small amount of language, we rely on contributors to help us with the translations.
+
+The translations process is:
+- While developing, new translation messages may be added or existing ones changed.
+- When a new version will releases soon, a maintainer will open an issue informing about this.
```
Comment (jasonblais) [inline CONTRIBUTING.md]:
```suggestion
- When a new version is planned to release soon, a maintainer opens an issue informing about this.
```

[the remark this answers is not in the document]
Comment (jasonblais) [reply CONTRIBUTING.md]: And I assume this is the repo maintainer, not translation maintainer?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+- The maintainer will ping all translation maintainer to inform them about this.
+- They open PR's with new translations, which (may) get reviewed by other translators.
```
Comment (jasonblais) [inline CONTRIBUTING.md]:
```suggestion
- Translation maintainers submit PRs with new translations, which may get reviewed by other translators.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+- After all translation PR's are merged, the new version is released.
```
Comment (jasonblais) [inline CONTRIBUTING.md]:
```suggestion
- After all translation PRs are merged, the new version is released.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+### Translation Maintainers
+
+- France: [@ldidry](https://github.com/ldidry)
+- German: [@hanzei](https://github.com/hanzei)
+- Japanese: [@kaakaa](https://github.com/kaakaa/)
+
+### Translation Process for Existing Languages
+
+1. Ensure all translation messages are correctly extracted:
```
Comment (kaakaa) [inline CONTRIBUTING.md]: It's better to add the process of installing `go-i18n`, or add new section about requirements for translating.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+`goi18n merge -format json -outdir assets/i18n/ assets/i18n/active.*.json assets/i18n/translate.*.json`
+
+5. Commit **only the language files you touched** and [submit a PR](https://github.com/matterpoll/matterpoll/compare).
+
+### Translation Process for New Languages
+
+Let's say you want to translate the local `de`. Replace  `de` in the following commands with the local you want to translate. See [here](https://github.com/mattermost/mattermost-server/tree/master/i18n) for the list of possible locals.
+
+1. Create a translation file:
```
Comment (kaakaa) [inline CONTRIBUTING.md]: Same is here.

File: CONTRIBUTING.md (hunk 4 above)
Comment (hanzei) [reply CONTRIBUTING.md]: Added a paragraph about this.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+
+## Helping with translations
+
+Matterpoll supports localization in various languages. Because we as the maintainer only speak a small amount of language, we rely on contributors to help us with the translations.
+
+The translations process is:
+- While developing, new translation messages may be added or existing ones changed.
+- When a new version will releases soon, a maintainer will open an issue informing about this.
+- The maintainer will ping all translation maintainer to inform them about this.
```
[the remark this answers is not in the document]
Comment (hanzei) [reply CONTRIBUTING.md]: Made this more clear.

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (hanzei) [reply CONTRIBUTING.md]: > Is there an example issue?

No. Currently the process is just defined, but we havn't tried it out.


> And I assume this is the repo maintainer, not translation maintainer?

Correct. :+1: I updated the text to make this more clear.

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hanzei) [reply CONTRIBUTING.md]: Currently not.

File: CONTRIBUTING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (hanzei) [reply CONTRIBUTING.md]: Good question.

We have no hard deadline when we want to release a version. To we can wait "a bit". But if no PR is submitted, we release without it.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+`goi18n merge -format json -outdir assets/i18n/ assets/i18n/active.en.json assets/i18n/translate.de.json`
+
+Translate all messages in `asserts/i18n/translate.de.json` and rename it to `active.de.json`.
+
+[Submit a PR](https://github.com/matterpoll/matterpoll/compare) with this file and add you to the list of [Translation Maintainer](#translation-maintainer)
+
+
+## Submitting Patches
+
+If you contributing a feature, [please open a feature request](https://github.com/matterpoll/matterpoll/issues/new?template=feature_request.md) first. This way the feature can be discussed and fully specified before you start working on this. Small code changes can be submitted without opening an issue first.
```
[the remark this answers is not in the document]
Comment (hanzei) [reply CONTRIBUTING.md]: > I wonder what defines as a "small" issue. We could use a guideline referred to here developers.mattermost.com/contribute/getting-started/contributions-without-ticket

I see your point. But for now this is fine. There hasn't been a lot of contributions jet. We can improve the definition if this becomes a problem.

**Comment by hanzei** [comment]
@jasonblais @kaakaa Please take a second look.

**Comment by hanzei** [comment]
@jasonblais I don't see you note

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+Matterpoll supports localization to various languages. We as maintainers rely on contributors to help with the translations.
+
+Matterpoll uses [go-i18n](https://github.com/nicksnyder/go-i18n) as library and tool to manage translation. The CLI tool `goi18n` is required to manage translation. You can install it by running `env GO111MODULE=off go get -u github.com/nicksnyder/go-i18n/v2/goi18n`.
+
+The localization process is defined below:
+- During development, new translation strings may be added or existing ones updated.
+- When a new version is planned to release soon, a repository maintainer opens an issue informing about this. The maintainer will ping all translation maintainer to inform them about this.
+- Translation maintainers submit PRs with new translations, which may get reviewed by other translators.
+- After all translation PRs are merged, the new version is released.
```
Comment (jasonblais) [inline CONTRIBUTING.md]: @hanzei It was about the translations PRs and how long we'd wait for translation maintainers to submit them

Comment (hanzei) [reply CONTRIBUTING.md]: Please see my answer at https://github.com/matterpoll/matterpoll/pull/166#discussion_r295316022

File: CONTRIBUTING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaakaa) [reply CONTRIBUTING.md]: I totally agree with @hanzei. If some messages are not translated, it's just be displayed in English and it's not critical bug.

And I'm wondering that `a bit` is about a week.

[the remark this answers is not in the document]
Comment (hanzei) [reply CONTRIBUTING.md]: Update the text to reflect this

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+Matterpoll supports localization to various languages. We as maintainers rely on contributors to help with the translations.
+
+Matterpoll uses [go-i18n](https://github.com/nicksnyder/go-i18n) as library and tool to manage translation. The CLI tool `goi18n` is required to manage translation. You can install it by running `env GO111MODULE=off go get -u github.com/nicksnyder/go-i18n/v2/goi18n`.
+
+The localization process is defined below:
+- During development, new translation strings may be added or existing ones updated.
+- When a new version is planned to release soon, a repository maintainer opens an issue informing about this. The maintainer will ping all translation maintainer to inform them about this.
+- Translation maintainers submit PRs with new translations, which may get reviewed by other translators.
+- After all translation PRs are merged, the new version is released. If a translation PR is not submitted within in week, the release will be cut anyway.
```
Comment (jasonblais) [inline CONTRIBUTING.md]:
```suggestion
- After all translation PRs are merged, the new version is released. If a translation PR is not submitted within a week, the release is cut without it.
```

Comment (jasonblais) [reply CONTRIBUTING.md]: Just minor proposal, otherwise LGTM!

**Comment by hanzei** [comment]
Thanks for you help @jasonblais :+1:

**Pull request merged**